api/lib/domain/usecases/get-user-certification-eligibility.js

```diff
@@ -14,6 +49,157 @@ const getUserCertificationEligibility = async function ({
+      complementaryCertificationBadgesHistories,
+    });
+
+    return _.chain(complementaryVersioning.complementaryCertificationBadges)
+      .filter(({ deactivationDate }) => !!deactivationDate)
+      .groupBy('deactivationDate')
+      .values()
+      .value();
+  }
+
+  return [];
+};
+
+/**
+ * @param {Object} params
+ * @param {number} params.stillValidBadgeAcquisitionComplementaryCertificationBadgeId
+ * @param {Array<number>} params.outdatedVersions
+ * @returns {Boolean}
+ */
+const _isOutDatedBadgeAcquisitionByMoreThanOneVersion = ({
+  stillValidBadgeAcquisitionComplementaryCertificationBadgeId,
+  outdatedVersions,
+}) => {
+  if (!outdatedVersions.length) {
+    return false;
+  }
+  const lastOutedVersion = outdatedVersions[0];
+  const isBadgeAcquisitionFoundInPreviousVersion = lastOutedVersion.some(
+    ({ id }) => id === stillValidBadgeAcquisitionComplementaryCertificationBadgeId,
+  );
+  return outdatedVersions.length > 1 && !isBadgeAcquisitionFoundInPreviousVersion;
+};
+
+/**
+ * @param {Object} params
+ * @param {number} params.complementaryCertificationId
+ * @param {Array<ComplementaryCertificationBadgesHistory>} params.complementaryCertificationBadgesHistories
+ * @returns {ComplementaryCertificationVersioning}
+ */
+const _toComplementaryCertificationVersioning = ({
+  complementaryCertificationId,
+  complementaryCertificationBadgesHistories,
+}) => {
+  const complementaryCertificationBadges = complementaryCertificationBadgesHistories.flatMap(
+    (complementaryCertificationBadgesHistory) => {
+      return complementaryCertificationBadgesHistory.complementaryCertificationBadgeIds.map(
+        (complementaryCertificationBadgeId) => {
+          return new ComplementaryCertificationBadge({
+            ...complementaryCertificationBadgesHistory,
+            id: complementaryCertificationBadgeId,
+          });
+        },
+      );
+    },
+  );
+
+  return new ComplementaryCertificationVersioning({ complementaryCertificationId, complementaryCertificationBadges });
+};
+
+/**
+ * @param {Object} params
+ * @param {Array<Object>} params.targetProfileHistory
+ * @returns {ComplementaryCertificationBadgesHistory}
+ */
+function _getComplementaryCertificationBadgesHistories({ targetProfileHistory }) {
+  return targetProfileHistory.map((complementaryCertificationBadge) => {
+    const complementaryCertificationBadgeIds = complementaryCertificationBadge.badges.map(
+      ({ complementaryCertificationBadgeId }) => complementaryCertificationBadgeId,
+    );
+
+    return new ComplementaryCertificationBadgesHistory({
+      complementaryCertificationBadgeIds,
+      activationDate: complementaryCertificationBadge.attachedAt,
+      deactivationDate: complementaryCertificationBadge.detachedAt,
+    });
+  });
+}
+
+async function _getTargetProfileHistory({ targetProfileHistoryRepository, certifiableBadgeAcquisition }) {
+  const currentsTargetProfileHistoryWithBadgesByComplementaryCertification =
```

Comment:
```suggestion
  const currentTargetProfilesHistoryWithBadges =
```
En me fiant au nom de la méthode de repository, ce devrait plutôt être ce nom ci, non ?

Reply:
(non - si - non, 🎁)

Reply:
ou
```suggestion
  const targetProfilesHistoryWithBadgesForCurrentComplementaryCertification =
```
🤷